services/cd-service/pkg/repository/transformer_db_test.go

```diff
@@ -3884,14 +3852,59 @@ func TestDeleteEnvironment(t *testing.T) {
 			ctx := testutil.MakeTestContext()
 			r := repo.(*repository)
 			err := r.State().DBHandler.WithTransaction(ctx, false, func(ctx context.Context, transaction *sql.Tx) error {
-				_, _, _, err := repo.ApplyTransformersInternal(testutil.MakeTestContext(), transaction, tc.Transformers...)
+				_, state, _, err := repo.ApplyTransformersInternal(testutil.MakeTestContext(), transaction, tc.Transformers...)
 				if err != nil {
 					return err
 				}
+
```

Comment:
to be a bit closer to how it runs on production, 
there should be a new transaction here.

```suggestion
err := r.State().DBHandler.WithTransaction(ctx, false, func(ctx context.Context, transaction *sql.Tx) error {
// ...
```